src/ipips/ipip-0476.md

```diff
@@ -0,0 +1,145 @@
+---
+title: "IPIP-0476: Delegated Routing DHT Closest Peers API"
+date: 2025-08-19
+ipip: proposal
+editors:
+  - name: Alex Potsides
+    github: achingbrain
+    affiliation:
+      name: Shipyard
+      url: https://ipshipyard.com
+  - name: Marcin Rataj
+    github: lidel
+    affiliation:
+      name: Shipyard
+      url: https://ipshipyard.com
+relatedIssues:
+  - https://github.com/ipfs/specs/pull/476
+  - https://github.com/ipfs/specs/pull/497
+order: 476
+tags: ['ipips']
+---
+
+## Summary
+
+Add a new HTTP endpoint to the Delegated Routing API that allows clients to find the closest peers to a given peer ID without being full DHT clients.
+
+## Motivation
+
+Browser nodes and other resource-constrained clients need to perform peer discovery operations without the overhead of being full DHT clients. The primary use case is for browser nodes performing random walks to find peers that they can make circuit relay reservations on.
+
+Currently, to find peers close to a particular key in the DHT keyspace, a node must:
+1. Be a full DHT client with all the associated overhead
+2. Maintain connections to many peers
```

Comment:
In theory this isn't necessary.

It is true that many connections are necessary to perform a lookup, but a client technically doesn't need to maintain any connection when not performing a lookup actively. It should remember at least some addresses of DHT servers (e.g bootstrappers are enough).

I think we just miss light DHT client implementations.